### DISCUSSION TITLE
'std::runtime_error'   what():  Joint name sizes do not match!

### DISCUSSION THREAD

**Discussion by mohamedsayed18** [tesseract-robotics/tesseract_planning#88] [Q&A] [answered]
I'm trying to add an external axis. but I get this error and I don't know how to fix it.
```
terminate called after throwing an instance of 'std::runtime_error'
 what():  Joint name sizes do not match!
```
I did as follows and I paid attention that the `joints_names` and `joint_start_pos` otherwise it throws an error that they are not equal
```
joint_names.push_back("positioner_base_joint");
joint_names.push_back("positioner_joint_1");
```
```
joint_start_pos(6) = 0;
joint_start_pos(7) = 0;
```
Any help is appreciated

**Comment by Levi-Armstrong** [comment, marked as answer]
Just wanted to check if you resolved the issues? If not are you modifying one of the examples?

**Reply by mohamedsayed18** [reply]
Still trying to figure it out. it is a new example using the cartesian_example as my base

**Reply by Levi-Armstrong** [reply]
What issues are you running into? Also, recommend taking a look at the puzzle_piece_auxillary_axes_example if you have not already.

**Reply by mohamedsayed18** [reply]
I looked at the puzzle auxiliary as a reference. In order to add the external axis(i used the one from tesseract_ros_workcell). 
I added the joint names to the `Joint_names` and set their values in the `joint_start_pos` vector. but the node terminate with the following error.
```
setting /run_id to 7e0ce20a-c2e2-11eb-bac9-2d8ccb16d0b8
process[rosout-1]: started with pid [17093]
started core service [/rosout]
process[virtual_joint_broadcaster_0-2]: started with pid [17100]
process[ABB_Node-3]: started with pid [17106]
process[tesseract_ros_examples_freespace_hybrid_example_rviz-4]: started with pid [17107]
terminate called after throwing an instance of 'std::runtime_error'
  what():  Joint name sizes do not match!
[ABB_Node-3] process has died [pid 17106, exit code -6, cmd /media/mohamed/C03CCDB43CCDA62E/work/Mx3D/mx_space/devel/lib/abb_example/abb_example_abb_planner_example_node __name:=ABB_Node __log:=/home/mohamed/.ros/log/7e0ce20a-c2e2-11eb-bac9-2d8ccb16d0b8/ABB_Node-3.log].
log file: /home/mohamed/.ros/log/7e0ce20a-c2e2-11eb-bac9-2d8ccb16d0b8/ABB_Node-3*.log
```
I also noticed that if I added the names to the `joint_names`, and for the joint values `joint_start_pos` i just change the size of the vector and I didn't push the two new joints(external axes joints) to the vector I face the same error

**Reply by mohamedsayed18** [reply]
I think the error was caused as I was referring to `manipulator` instead of `manipulator_aux` from the srdf file.
```
CompositeInstruction program("FREESPACE", CompositeInstructionOrder::ORDERED, ManipulatorInfo("manipulator_aux"));
```

```
    <group name="manipulator">
        <chain base_link="base_link" tip_link="tool0" />
    </group>

    <group name="positioner_only">
        <chain base_link="positioner_base_link" tip_link="positioner_tool0"/>
    </group>

    <group name="manipulator_aux">
        <joint name="joint_1" />
        <joint name="joint_2" />
        <joint name="joint_3" />
        <joint name="joint_4" />
        <joint name="joint_5" />
        <joint name="joint_6" />
        <joint name="positioner_base_joint" />
        <joint name="positioner_joint_1" />
    </group>
```
The error is not showing anymore but the program still terminates and the following warning is showing
```
Warning: Joint groups are currently not supported by inverse kinematics!
```

**Reply by Levi-Armstrong** [reply]
Is the robot attached to the positioner or is it an external two-axis positioner?

**Reply by mohamedsayed18** [reply]
It is an external two axis positioner I used the one in tesseract workcell

**Reply by Levi-Armstrong** [reply]
Do you have this in a public repository where I could take a look?

**Reply by mohamedsayed18** [reply]
You can have a look [here](https://github.com/mohamedsayed18/motion_planner). Thank you very much.

**Reply by Levi-Armstrong** [reply]
I took a look and noticed a few things I would look into. I will need to pull down and test but I recommend trying to get your example to work with just the robot and exclude the external positioner. Since you are doing cartesian planning, I recommend leveraging OPW kinematics and the custom IK solver for an industrial robot with an external positioner. Both can be added to the srdf. I have added the documentation link which we just started updated and will be making additions over the next months.

https://tesseract-docs.readthedocs.io

**Reply by mohamedsayed18** [reply]
My example is working fine if I define only the manipulator joints in the `joint_names`, `joint_start_pos` and also in the ManipulatorInfo.
If I used the `manipulator_aux` which the external joints are defined in the srdf, I got an error.

**Reply by Levi-Armstrong** [reply]
I have created a [PR](https://github.com/mohamedsayed18/motion_planner/pull/1) against your repository.

**Reply by Levi-Armstrong** [reply]
I have also created a [PR](https://github.com/ros-industrial-consortium/tesseract/pull/610) to fix issues in ROP and REP parsers.

**Reply by mohamedsayed18** [reply]
Do we need to open a PR for changes in trajopt_ros
https://github.com/mohamedsayed18/motion_planner/blob/13922b17b290675f0d6ca43736f05b2a546f22fe/trajopt_ros/trajopt_sco/CMakeLists.txt#L37

**Reply by Levi-Armstrong** [reply]
I don't think so but I am looking into why CI is having issues copying the files for running tests all of a sudden.
